src/store/diagnostics.ts

```diff
@@ -97,12 +118,23 @@ export const actions = {
 
     rowToUpdate.mute = isMuted;
 
+    const mutedChecks = { ...rootState.preferences.preferences.diagnostics.mutedChecks, [rowToUpdate.id]: isMuted };
+
+    await dispatch(
+      'preferences/commitPreferences',
+      {
+        ...rootState.credentials.credentials as ServerState,
+        payload: { diagnostics: { mutedChecks } } as RecursivePartial<Settings>,
```

Comment:
Just realized that `{ diagnostics: { mutedChecks: { [rowToUpdate.id]: isMuted } } }` would work better, but this is probably fine.